modules/fun.js

```diff
@@ -0,0 +1,450 @@
+async function slashFunRoll(int) {
+  let dice = int.options.getInteger('dice');
+  let sides = int.options.getInteger('sides');
+  let modifier = int.options.getInteger('modifier');
+  if (!dice) dice = 1;
+  if (!sides) sides = 6;
+  if (!modifier) modifier = 0;
+  /** @type {string[][]} */
+  const rolls = [];
+  let total = 0;
+  rolls[sides] = [];
+  const num = dice;
+  if (num && num <= 10000) {
+    for (let i = 0; i < num; i++) {
+      const val = Math.ceil(Math.random() * sides);
+      rolls[sides].push((i == 0 ? `**d${sides}:** ` : ``) + val);
+      total += val;
+    }
+  } else {
+    return int.editReply(`I'm not going to roll *that* many dice... 🙄`);
+  }
+  if (modifier) {
+    total += dice;
+    rolls[sides].push(`**${dice}**`);
+  }
+  if (rolls.length > 0) {
+    return int.editReply(`You rolled ${dice}d${sides} and got:${total}\n` +
+        ((rolls.reduce((a, c) => a + c.length, 0) > 20) ? `` : ` ( ${rolls.reduce((a, c) => a + c.join(`, `) + `; `, ``)})`));
+  } else {
+    return int.editReply(`you didn't give me anything to roll.`);
+  }
+}
+/**
+ * function ball8
+ * @param {Discord.ChatInputCommandInteraction} int a /fun 8ball interaction
+ */
+async function slashFun8ball(int) {
+  const question = int.options.getString(`question`);
+  if (!question || !question.endsWith(`?`)) {
+    return int.editReply(`you need to ask me a question, silly.`);
+  } else {
+    const outcomes = [
+      `It is certain.`,
+      `It is decidedly so.`,
+      `Without a doubt.`,
+      `Yes - definitely.`,
+      `You may rely on it.`,
+      `As I see it, yes.`,
+      `Most likely.`,
+      `Outlook good.`,
+      `Yes.`,
+      `Signs point to yes.`,
+      // `Reply hazy, try again.`,
+      // `Ask again later.`,
+      // `Better not tell you now.`,
+      // `Cannot predict now.`,
+      // `Concentrate and ask again.`,
+      `Don't count on it.`,
+      `My reply is no.`,
+      `My sources say no.`,
+      `Outlook not so good.`,
+      `Very doubtful.`
+    ];
+    return int.editReply(`You asked :"${question}"\n` +
+      `The 8ball replies:\n` +
+      u.rand(outcomes));
+  }
+}
+/**
+ * function repost
+ * @param {Discord.ChatInputCommandInteraction} int a /fun repost interaction
+ */
+async function slashFunRepost(int) {
+  if (!int.channel) {
+    return int.editReply(`I don't know where here is, so I can't find anything to repost... try in a more normal channel.`);
+  }
+  const latest = (await int.channel.messages.fetch({ limit: 100 })).filter(m => m.attachments.size > 0).last();
+  if (!latest) {
+    return int.editReply(`I couldn't find anything in the last 100 messages to repost.`);
+  }
+  const imgToRepost = latest.attachments.first().url;
+  return int.editReply(imgToRepost);
+}
+/**
+ * function buttermelon
+ * @param {Discord.ChatInputCommandInteraction} int a /fun buttermelon interaction
+ */
+async function slashFunButtermelon(int) {
+  const buttermelonFacts = require('../data/buttermelonFacts.json');
```

Comment:
at the top, also `require("...").facts` (or just make the file an array)

Reply:
ok, I did not move this to the top with the others because it seems very function specific, but if you say so!